File: citas/src/main/java/com/hospital/p_citas/citas/entidades/Cita.java
package com.hospital.p_citas.citas.entidades;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Getter
@Setter
@ToString
@Table(name = "citas")
public class Cita {

    @Id
    private String id;

    private long idmedico;

    private long idpaciente;

    private long idhabitacion;

    private String fecha;

    private String hora;

}

File: front-end/src/pages/citas/Cita.ts
import Habitacion from "../habitaciones/Habitacion";
import Medico from "../medicos/Medico";
import Paciente from "../pacientes/Paciente";

//Cita
interface Cita {
  id: string;
  idmedico: string;
  medico?: Medico;
  idpaciente: string;
  paciente?: Paciente;
  idhabitacion: string;
  habitacion?: Habitacion;
  fecha: string;
  hora: string;
}

export default Cita;
